job-runner/chain/chain.go

```diff
@@ -124,6 +124,9 @@ func (c *Chain) IsDoneRunning() (done bool, complete bool) {
 			}
 			// This job is pending but not runnable which means a previous job
 			// failed.
+		case proto.STATE_UNKNOWN:
+			// Treat unknown as a failure to retry, is possible.
+			fallthrough
 		case proto.STATE_FAIL:
```

Comment:
you can have multiple items in a case statement, so I'd switch the fallthrough for `case proto.STATE_FAIL, proto.STATE_UNKNOWN:` just because using fallthrough is pretty unusual in Go

Reply:
Done.